### ISSUE TITLE
Opcode for 6.38

### ISSUE THREAD

**Issue by aoiszz** [Soreepeong/XivAlexander#605]

**Comment by skhreis** [comment]
```
{
	"C2S_ActionRequest": "0x029e",
	"C2S_ActionRequestGroundTargeted": "0x029e",
	"PatchCode": [],
	"S2C_ActionEffect01": "0x00fb",
	"S2C_ActionEffect08": "0x02ef",
	"S2C_ActionEffect16": "0x03c6",
	"S2C_ActionEffect24": "0x0097",
	"S2C_ActionEffect32": "0x024b",
	"S2C_ActorCast": "0x015f",
	"S2C_ActorControl": "0x02c2",
	"S2C_ActorControlSelf": "0x0256",
	"Server_IpRange": "80.239.145.0/24, 124.150.157.0/24, 153.254.80.0/24, 202.67.52.0/24, 204.2.229.0/24, 210.94.0.0/24",
	"Server_PortRange": "1025-65535"
}
```

This is what I got right now, it's not official and it's my first time doing it, but it looks like everything is working just fine, it does not look like its showing on the logger properly for me, but in game I'm able to do the norm of what I do with xivalex, worth a shot.

For those who know have experience getting opcodes, I do not see any C2S_ActionRequest for normal actions, they all show up as ActionRequestGroundTargeted with the same OPCode, its working for me so idk what to say 🤷 it is different than normal

**Comment by phanlongtran** [comment]
How do i use the OPCode? I "Edit OPCode" option then paste it in and save and try restarting the game multiple times but it does not work. I even replace the ones in the configuration folder with your new code but it still clip. I also tried to get the code on my own but all it show is "S2C_Raw: Error: OodleNetwork1UDP_Decode error". Tried reinstalling alex multiple times but still it persist.

**Comment by SinisterSedna** [comment]
how do you open up the menu so i can change the opcodes? ever since the update despite even reinstalling the program its still not showing up in my taskbar

**Comment by phanlongtran** [comment]
Try completely remove alex and all of the config folder then install it on ffxivboot.exe in boot folder and use that boot to start the game instead pf XIVLauncher cause it's disabled now.

**Comment by SinisterSedna** [comment]
oh duh! suppose that makes sense! i didn't think of that thank you so much!

**Comment by phanlongtran** [comment]
If you can get the OPCode working come back and update it here how you did it. Thanks in advanced.

**Comment by skhreis** [comment]
To update the opcodes there are two methods.

A) Find the xivalex folder and look for the file that has JP on it. Open it with notepad and copy paste the opcodes there and save 

B) In xivalex, NOT the log, but the other window, go to the configure tab and hit on edit opcodes configuration and copy paste and then save.

Make sure when you are editing the opcodes copy paste the entirety of the text including the brackets(Ctrl+A is a keybind for that) and paste the entirety of the text including the brackets to ensure you don't overlap the brackets.

**Comment by skhreis** [comment]
Until there is an official update this is like a bandaid fix and you may run into minor errors in the log but as long as there is no red or yellow text in the logger it should be working fine. Or at least that's how it did with me. I'm not at my PC to give a snippet but it's like it's attempting to and assigning the action based off the action id(movement, spell, anything u do rly) and is in normal black text. This is my first time finding and doing this so I have no idea what exactly am I suppose to see but it works so 🤷 if anyone is more experienced with this can comment and explain more or even fix it in that regard but because there was no C2S_ActionRequest in my search for the opcodes SE might have changed it with the patch?

**Comment by Spanglish1337** [comment]
@skhreis Does this mean that JP has same OPCodes as NA? I thought every region had diff opcodes or something like that

**Comment by skhreis** [comment]
> @skhreis Does this mean that JP has same OPCodes as NA? I thought every region had diff opcodes or something like that

Yes, NA/EU/JP all share similar opcodes the ips are there within the opcodes. Server IP range and port range

**Comment by thirdpartytool** [comment]
```
{
    "C2S_ActionRequest": "0x029e",
    "C2S_ActionRequestGroundTargeted": "0x0205",
    "Common_UseOodleTcp": true,
    "PatchCode": [],
    "S2C_ActionEffect01": "0x00fb",
    "S2C_ActionEffect08": "0x02ef",
    "S2C_ActionEffect16": "0x03c6",
    "S2C_ActionEffect24": "0x0097",
    "S2C_ActionEffect32": "0x024b",
    "S2C_ActorCast": "0x015f",
    "S2C_ActorControl": "0x02c2",
    "S2C_ActorControlSelf": "0x0256",
    "Server_IpRange": "80.239.145.0/24, 124.150.157.0/24, 153.254.80.0/24, 202.67.52.0/24, 204.2.29.0/24, 210.94.0.0/24",
    "Server_PortRange": "1025-65535"
}
```

the above opcodes, but now the `C2S_ActionRequestGroundTargeted` *should* (keyword) be right, and the new IP addresses for the north american data centers are put into `Server_IpRange`, i make 0 guarantee that it'll work perfectly, i just know that i hit a training dummy and the funny log lines come out telling me it was doing stuff

**Comment by pepper-jelly** [comment]
> ```
> {
>     "C2S_ActionRequest": "0x029e",
>     "C2S_ActionRequestGroundTargeted": "0x0205",
>     "Common_UseOodleTcp": true,
>     "PatchCode": [],
>     "S2C_ActionEffect01": "0x00fb",
>     "S2C_ActionEffect08": "0x02ef",
>     "S2C_ActionEffect16": "0x03c6",
>     "S2C_ActionEffect24": "0x0097",
>     "S2C_ActionEffect32": "0x024b",
>     "S2C_ActorCast": "0x015f",
>     "S2C_ActorControl": "0x02c2",
>     "S2C_ActorControlSelf": "0x0256",
>     "Server_IpRange": "80.239.145.0/24, 124.150.157.0/24, 153.254.80.0/24, 202.67.52.0/24, 204.2.29.0/24, 210.94.0.0/24",
>     "Server_PortRange": "1025-65535"
> }
> ```
> 
> the above opcodes, but now the `C2S_ActionRequestGroundTargeted` _should_ (keyword) be right, and the new IP addresses for the north american data centers are put into `Server_IpRange`, i make 0 guarantee that it'll work perfectly, i just know that i hit a training dummy and the funny log lines come out telling me it was doing stuff

thanks, it works on ps5 as well with [XivMitmLatencyMitigator](https://github.com/Soreepeong/XivMitmLatencyMitigator), but requires overwrite changes every time until repo updated with new codes

**Comment by Arkzera** [comment]
> ```
> {
>     "C2S_ActionRequest": "0x029e",
>     "C2S_ActionRequestGroundTargeted": "0x0205",
>     "Common_UseOodleTcp": true,
>     "PatchCode": [],
>     "S2C_ActionEffect01": "0x00fb",
>     "S2C_ActionEffect08": "0x02ef",
>     "S2C_ActionEffect16": "0x03c6",
>     "S2C_ActionEffect24": "0x0097",
>     "S2C_ActionEffect32": "0x024b",
>     "S2C_ActorCast": "0x015f",
>     "S2C_ActorControl": "0x02c2",
>     "S2C_ActorControlSelf": "0x0256",
>     "Server_IpRange": "80.239.145.0/24, 124.150.157.0/24, 153.254.80.0/24, 202.67.52.0/24, 204.2.29.0/24, 210.94.0.0/24",
>     "Server_PortRange": "1025-65535"
> }
> ```
> 
> the above opcodes, but now the `C2S_ActionRequestGroundTargeted` _should_ (keyword) be right, and the new IP addresses for the north american data centers are put into `Server_IpRange`, i make 0 guarantee that it'll work perfectly, i just know that i hit a training dummy and the funny log lines come out telling me it was doing stuff

Working on linux version, tyvm

**Comment by DiegoSalazar231** [comment]
> > ```
> > {
> >     "C2S_ActionRequest": "0x029e",
> >     "C2S_ActionRequestGroundTargeted": "0x0205",
> >     "Common_UseOodleTcp": true,
> >     "PatchCode": [],
> >     "S2C_ActionEffect01": "0x00fb",
> >     "S2C_ActionEffect08": "0x02ef",
> >     "S2C_ActionEffect16": "0x03c6",
> >     "S2C_ActionEffect24": "0x0097",
> >     "S2C_ActionEffect32": "0x024b",
> >     "S2C_ActorCast": "0x015f",
> >     "S2C_ActorControl": "0x02c2",
> >     "S2C_ActorControlSelf": "0x0256",
> >     "Server_IpRange": "80.239.145.0/24, 124.150.157.0/24, 153.254.80.0/24, 202.67.52.0/24, 204.2.29.0/24, 210.94.0.0/24",
> >     "Server_PortRange": "1025-65535"
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > the above opcodes, but now the `C2S_ActionRequestGroundTargeted` _should_ (keyword) be right, and the new IP addresses for the north american data centers are put into `Server_IpRange`, i make 0 guarantee that it'll work perfectly, i just know that i hit a training dummy and the funny log lines come out telling me it was doing stuff
> 
> thanks, it works on ps5 as well with [XivMitmLatencyMitigator](https://github.com/Soreepeong/XivMitmLatencyMitigator), but requires overwrite changes every time until repo updated with new codes
> 
> definitions.json for na servers

Hey. I´m also playing on ps5, but I´m not sure how to update the file or where to even find definitions.json. Could you please explain how you make it work again? Or maybe if you know a site tutorial like where I can get more info about it I´ll highly appreciate.

**Comment by Gianluca90** [comment]
> ```
> {
>     "C2S_ActionRequest": "0x029e",
>     "C2S_ActionRequestGroundTargeted": "0x0205",
>     "Common_UseOodleTcp": true,
>     "PatchCode": [],
>     "S2C_ActionEffect01": "0x00fb",
>     "S2C_ActionEffect08": "0x02ef",
>     "S2C_ActionEffect16": "0x03c6",
>     "S2C_ActionEffect24": "0x0097",
>     "S2C_ActionEffect32": "0x024b",
>     "S2C_ActorCast": "0x015f",
>     "S2C_ActorControl": "0x02c2",
>     "S2C_ActorControlSelf": "0x0256",
>     "Server_IpRange": "80.239.145.0/24, 124.150.157.0/24, 153.254.80.0/24, 202.67.52.0/24, 204.2.29.0/24, 210.94.0.0/24",
>     "Server_PortRange": "1025-65535"
> }
> ```
> 
> the above opcodes, but now the `C2S_ActionRequestGroundTargeted` _should_ (keyword) be right, and the new IP addresses for the north american data centers are put into `Server_IpRange`, i make 0 guarantee that it'll work perfectly, i just know that i hit a training dummy and the funny log lines come out telling me it was doing stuff

One of your IP is 204.2.29.0/24 , while in a previous post by [skhreis](https://github.com/skhreis) that IP is 204.2.229.0/24 ; what's the correct one?

**Comment by pepper-jelly** [comment]
@DiegoSalazar231 There must-read tutorial https://github.com/Soreepeong/XivMitmLatencyMitigator. I would add that it works in virtualbox too if you change default network to bridged adapter.
After setup you can run `curl https://pastebin.com/raw/jf66WP69 -o definitions.json` for now and then use mitigation.py as usual. Then on ps5 you need to change default gateway to ip of virtual machine - `ip addr`.

**Comment by DiegoSalazar231** [comment]
> @DiegoSalazar231 There must-read tutorial https://github.com/Soreepeong/XivMitmLatencyMitigator. I would add that it works in virtualbox too if you change default network to bridged adapter(also 204.2.229.0/24 now 204.2.29.0/24). After setup you can run `curl https://pastebin.com/raw/jf66WP69 -o definitions.json` for now and then use mitigation.py as usual. Then on ps5 you need to change default gateway to ip of virtual machine - `ip addr`.

I mean, I've already installed xivmitm a couple of months ago and worked perfectly, but I have the same error due patch 6.38. And like I've seen this too many times, I was trying to learn how to fix it as many of you guys does haha. Thank you for your reply, I'll try that line later <3 

Just one question, did you make yourself that url with the new codes in this thread to fix the now broken mitigate.py?

**Comment by pepper-jelly** [comment]
@DiegoSalazar231 Yes. mitigate.py not broken but there check if the file older than hour and then re-downloads default. So use `sudo touch definitions.json; sudo python mitigate.py` until #606 accepted to keep it working.

**Comment by phanlongtran** [comment]
@Gianluca90 29 would be the correct one as 204.2.29.x is the ip address for NA data center

**Comment by robyn0751** [comment]
> [skhreis](/skhreis)

Thank you for the opcodes. They work for me (EU player).

**Comment by phanlongtran** [comment]
> ```
> {
>     "C2S_ActionRequest": "0x029e",
>     "C2S_ActionRequestGroundTargeted": "0x0205",
>     "Common_UseOodleTcp": true,
>     "PatchCode": [],
>     "S2C_ActionEffect01": "0x00fb",
>     "S2C_ActionEffect08": "0x02ef",
>     "S2C_ActionEffect16": "0x03c6",
>     "S2C_ActionEffect24": "0x0097",
>     "S2C_ActionEffect32": "0x024b",
>     "S2C_ActorCast": "0x015f",
>     "S2C_ActorControl": "0x02c2",
>     "S2C_ActorControlSelf": "0x0256",
>     "Server_IpRange": "80.239.145.0/24, 124.150.157.0/24, 153.254.80.0/24, 202.67.52.0/24, 204.2.29.0/24, 210.94.0.0/24",
>     "Server_PortRange": "1025-65535"
> }
> ```
> 
> the above opcodes, but now the `C2S_ActionRequestGroundTargeted` _should_ (keyword) be right, and the new IP addresses for the north american data centers are put into `Server_IpRange`, i make 0 guarantee that it'll work perfectly, i just know that i hit a training dummy and the funny log lines come out telling me it was doing stuff

Thank you this worked for me (OCE player). It seems we are using .json file of JP.

**Comment by Gianluca90** [comment]
> 

@phanlongtran thanks for the answer, i guess the change in IP is because of the new servers on NA?
I have another question, what's the 210.94.0.0 IP? I managed to identify the others (for EU, NA, JP etc), but not that one.

**Comment by phanlongtran** [comment]
@Gianluca90 Seems to be Korea server address. I don't have access to that version of game so i can't confirm but it's most likely that since searching google for 210.94.0.0 returns in a bunch of South Korea IP websites.

**Comment by Gianluca90** [comment]
@phanlongtran ah i think you're right, i searched too and i found the same stuff as you. thanks

**Comment by pepper-jelly** [comment]
@Flaviorsb  make sure it's latest ffxiv_dx11.exe and open issue in XivMitmLatencyMitigator repo if there are still errors

**Comment by Flaviorsb** [comment]
> @Flaviorsb make sure it's latest ffxiv_dx11.exe and open issue in XivMitmLatencyMitigator repo if there are still errors

thanks, i installed my version of the game from steam and got ffxiv_dx11.exe from there. I tested it here now and it's working again.
